src/fs/path.ts

```diff
@@ -1,7 +1,8 @@
 import { NonEmptyPath } from './types'
 
+// segments
 export const splitParts = (path: string): string[] => {
-  return path.split('/').filter(p => p.length > 0)
+  return path.split('/').filter(p => p.length > 0) // why not `p == ''`
```

Comment:
either is fine!

Reply:
> Specifically, the > and < and >= and <= operators all operate either on two numeric values, or two strings, preferring numeric values. That is, if one value is a number, then the other is treated as a number. If a non-number can't be cleanly converted to a number (that is, if it ends up as NaN), then the result of the comparison is undefined. (That's a little problematic, because undefined will look like false in the context of an if statement.)

It's an edge case, but `===` is probably preferred.

```
'4' > 0
true
```

TS will stop this, but downstream people can call the SDK untyped

Reply:
ahh gotta love JS...

ok yup that makes sense :+1: